msu/utils/utils.nut

```diff
@@ -203,6 +203,48 @@
 	    return this.Timers[_id];
 	}
 
+	function benchmark( _functions, _iterations = 100000 )
+	{
+		// _functions can be a function to be benchmarked
+		// or an array of len 2 arrays where idx 0 is an id (can be string) and idx 1 is function to be benchmarked
+		if (typeof _functions == "function")
+			_functions = [["", _functions]];
+
+		local times = array(_functions.len());
+		local timer = ::MSU.Class.Timer("MSU_Benchmark");
+
+		foreach (i, entry in _functions)
+		{
+			local timeStart = timer.silentGet();
+			local f = entry[1];
+
+			for (local j = 0; j < _iterations; j++)
+			{
+				f();
+			}
+
+			local totalTime = timer.silentGet() - timeStart;
+
+			times[i] = {
+				TotalTime = totalTime,
+				TimePerIteration = totalTime / _iterations
+			};
+		}
+
+		::logWarning("MSU -- Benchmark with " + _iterations + " iterations");
+		foreach (i, entry in times)
+		{
+			local diff = entry.TimePerIteration - times[0].TimePerIteration;
+			local color = i == 0 ? "" : (diff <= 0 ? "green" : "red"); // the base function is printed without color, faster in green, slower in red
+
+			local timePerIterationText = format("<div style='display: inline-block; color:%s;'>%f (%s%f)</div>", color, entry.TimePerIteration, diff >= 0 ? "+" : "", diff);
+			diff = entry.TotalTime - times[0].TotalTime;
+			local totalTimeText = entry.TotalTime + (diff < 0 ? " (" : " (+") + diff + ")";
+			local timePctText = format("<div style='display: inline-block; color:%s'>%i%% %s</div>", color, ::Math.abs(100 * diff / times[0].TotalTime), diff < 0 ? "faster" : "slower");
+			::logInfo(format("<div style='display: inline-block;'>%s: %s ms (Total: %s ms) -- %s</div>", _functions[i][0] + "", timePerIterationText, totalTimeText, i == 0 ? "base time" : timePctText));
```

Comment:
instead of div style inline-block, this should be a `span` element, then you can remove that part.